aspnetcore/tutorials/min-web-api.md

```diff
@@ -0,0 +1,384 @@
+---
+title: "Tutorial: Create a minimal web API with ASP.NET Core"
+author: rick-anderson
+description: Learn how to build a minimal web API with ASP.NET Core.
+ms.author: riande
+ms.date: 8/24/2021
+monikerRange: '>= aspnetcore-6.0'
+no-loc: [Home, Privacy, Kestrel, appsettings.json, "ASP.NET Core Identity", cookie, Cookie, Blazor, "Blazor Server", "Blazor WebAssembly", "Identity", "Let's Encrypt", Razor, SignalR, Models]
+uid: tutorials/min-web-api
+---
+
+# Tutorial: Create a web minimal API with ASP.NET Core
+
+<!-- TODO: Remove aspnetcore\tutorials\min-web-api\samples\6.x -->
+By [Rick Anderson](https://twitter.com/RickAndMSFT)
+
+Minimal APIs are architected to create REST APIs with the minimal dependencies. They are ideal for microservices and apps that want to include only the minimum files, features, and dependencies in ASP.NET Core.
+
+This tutorial teaches the basics of building a miminal web API with ASP.NET Core. For a tutorial on creating a web API project that contains more features, see [Create a web API](xref:tutorials/first-web-api)
+
+## Overview
+
+This tutorial creates the following API:
+
+|API | Description | Request body | Response body |
+|--- | ---- | ---- | ---- |
+|`GET /` | Browser test, "Hello World" | None | Hello World!|
+|`GET /todoitems` | Get all to-do items | None | Array of to-do items|
+|`GET /todoitems/complete` | Get completed to-do items | None | Array of to-do items|
+|`GET /todoitems/{id}` | Get an item by ID | None | To-do item|
+|`POST /todoitems` | Add a new item | To-do item | To-do item |
+|`PUT /todoitems/{id}` | Update an existing item &nbsp; | To-do item | None |
+|`DELETE /todoitems/{id}` &nbsp; &nbsp; | Delete an item &nbsp; &nbsp; | None | None|
+
+## Prerequisites
+
+# [Visual Studio](#tab/visual-studio)
+
+[!INCLUDE[](~/includes/net-prereqs-vs-6.0.md)]
+
+# [Visual Studio Code](#tab/visual-studio-code)
+
+[!INCLUDE[](~/includes/net-prereqs-vsc-6.0.md)]
+
+<!-- add VS Mac later
+# [Visual Studio for Mac](#tab/visual-studio-mac)
+
+[!INCLUDE[](~/includes/net-prereqs-mac-6.0.md)]
+-->
+---
+
+## Create an empty web project
+
+# [Visual Studio](#tab/visual-studio)
+
+* Start Visual Studio 2022 and select **Create a new project**
+  ![Visual Studio new project start page](min-web-api/_static/create.png)
+* In the **Create a new project** dialog:
+  * Enter `Empty` in the **Search for templates** search box.
+  * Select the **ASP.NET Core Empty** template and select **Next**.
```

Comment:
Given this is a tutorial about API and not app, shouldn't we be using Web API template instead?

Reply:
@timheuer  the Web API template generates code for the traditional controller Web API. It creates a controller,  a WeatherForecast class/file,  and alot of code that's not used in this tutorial. The  traditional API's are not minimal, they use routing and a controller.

 The empty template creates a minimal API with the following code:

```
var builder = WebApplication.CreateBuilder(args);
var app = builder.Build();

app.MapGet("/", () => "Hello World!");

app.Run();
```

Note, I've removed the following code per https://github.com/dotnet/templating/issues/3885

```

if (app.Environment.IsDevelopment())
{
    app.UseDeveloperExceptionPage();
}
``

I hope to PR the templates soon and remove that redundant code. Even with that redundant code, it's far simpler than the traditional Web API and uses the same `MapGet` API.

I think the empty template is a much better starting point for this tutorial rather than having them remove a bunch of code and files.

In the [traditional Web API tutorial](https://docs.microsoft.com/en-us/aspnet/core/tutorials/first-web-api?view=aspnetcore-5.0&tabs=visual-studio), we accept the default ane generate swagger code, but then we have them delete the swagger code and refer them to the [Swagger tutorial](https://docs.microsoft.com/en-us/aspnet/core/tutorials/getting-started-with-nswag?view=aspnetcore-5.0&tabs=visual-studio). At this onboarding level, we've found it better to focus on API code and then let then graduate to Swagger, calling from JavaScript, etc.

Reply:
See: https://github.com/dotnet/aspnetcore/pull/36068

Reply:
> See: [dotnet/aspnetcore#36068](https://github.com/dotnet/aspnetcore/pull/36068)

OK, I can use that when it's publicly available.

Reply:
This will be in rc2.